k8s/plugin/src/main.rs

```diff
@@ -40,23 +62,27 @@ impl Deref for CliArgs {
 
 #[tokio::main]
 async fn main() {
-    let cli_args = CliArgs::args();
-    let _tracer_flusher = cli_args.init_tracing();
+    match CliArgs::args().await {
+        Ok(cli_args) => {
+            let _tracer_flusher = cli_args.init_tracing();
 
-    if let Err(error) = cli_args.execute().await {
-        let mut exit_code = 1;
-        match error {
-            Error::RestPlugin(error) => eprintln!("{error}"),
-            Error::RestClient(error) => {
-                eprintln!("Failed to initialise the REST client. Error {error}")
-            }
-            Error::Upgrade(error) => {
-                eprintln!("{error}");
-                exit_code = error.into();
+            if let Err(error) = cli_args.execute().await {
+                let mut exit_code = 1;
+                match error {
+                    Error::RestPlugin(error) => eprintln!("{error}"),
+                    Error::RestClient(error) => {
+                        eprintln!("Failed to initialise the REST client. Error {error}")
+                    }
+                    Error::Upgrade(error) => {
+                        eprintln!("{error}");
+                        exit_code = error.into();
+                    }
+                    Error::Generic(error) => eprintln!("{error}"),
+                }
+                std::process::exit(exit_code);
             }
-            Error::Generic(error) => eprintln!("{error}"),
         }
-        std::process::exit(exit_code);
+        Err(e) => eprintln!("{e}"),
```

Comment:
This needs to be flattened, it's not erroring out with -1

Reply:
I have added std::process::exit to error arm